### PR TITLE
Move to ESM

### PR DESCRIPTION
**Pull request by sindresorhus**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -104,15 +105,16 @@
 			"test/*"
 		],
 		"timeout": "1m",
-		"typescript": {
-			"compile": "tsc",
-			"rewritePaths": {
-				"test/": "dist/test/"
```
Comment (szmarczak) [inline package.json:110]: Is this not needed? `dist` is still referenced in `exports`.

Comment (sindresorhus) [reply package.json:110]: No, that config is just for tests. We now use `ts-node` instead which does this for us.

**Comment by sindresorhus** [comment]
I'm merging this to unblock other work, but let me know if there's anything else.

File: test/helpers/with-server.ts [hunk 2]
```diff
@@ -1,19 +1,19 @@
 import {promisify} from 'util';
-import * as test from 'ava';
+import {ExecutionContext, Macro} from 'ava'; // eslint-disable-line ava/use-test
```
Comment (szmarczak) [inline test/helpers/with-server.ts:2]: I also noticed that this is quite buggy. Does it also happen when doing `import type`?

Comment (sindresorhus) [reply test/helpers/with-server.ts:2]: We just need to fix the eslint-plugin-ava rule.

**Pull request merged**